### ISSUE TITLE
Prototype registration and login flow

### ISSUE THREAD

**Issue by blackjackkent** [new-xkit/pipeline#28]
* [ ] Create account
  * [ ] email
  * [ ] password
  * [ ] validate email does not exit
  * [ ] hash password
  * [ ] email verification (?)
* [ ] Log in
* [ ] Blank "main pipeline" skeleton to redirect to on successful login

**Comment by 0xazure** [comment]
Regarding email verification/validation: http://davidcel.is/posts/stop-validating-email-addresses-with-regex/

**Comment by Wolvan** [comment]
I agree, the only thing we should check for is "@" and ".", if we do checking at all, and that should just be a clientside check. If the mail doesn't come through, we simply deny feature access imo

**Comment by blackjackkent** [comment]
Makes sense to me. We definitely need to be sure they're not trying to sign up an email that's already in the DB though.

Do you think the verification email should be part of this task or separate?

**Comment by Wolvan** [comment]
It's part of registration, so it should be part of this PR. Do we want to have a ttl on the registraion mails?

**Comment by bvtsang** [comment]
Email verification should be part of this task because it's part of the registration process.

**Comment by Wolvan** [comment]
@bvtsang that if a person does not finalize the registration in time by clicking the confirmation link the registration entry gets deleted from the database and the email is free to be used for sign-up once more

**Comment by bvtsang** [comment]
Good point. Maybe the TTL should be a couple of hours?

With that said, if we do this then we should also have a "Resend verification email" process, just in case the email gets lost en-route.

**Comment by Wolvan** [comment]
TTL usually is 1-2 days, sometimes a week
